common/persistence/history_manager.go

```diff
@@ -532,16 +528,23 @@ func (m *executionManagerImpl) AppendHistoryNodes(
 	request *AppendHistoryNodesRequest,
 ) (*AppendHistoryNodesResponse, error) {
 
-	req, err := m.serializeAppendHistoryNodesRequest(ctx, request)
-
+	nodeReq, err := m.serializeAppendHistoryNodesRequest(request)
 	if err != nil {
 		return nil, err
 	}
 
-	err = m.persistence.AppendHistoryNodes(ctx, req)
+	err = m.persistence.AppendHistoryNodes(ctx, nodeReq)
```

Comment:
Append and Insert used to be in the same batch. Are you sure this won't cause any problems with them being in separate batches? For example, what happens if we crash after 536 and before 542?

Reply:
This failure condition (with nodes being written first with tree possibly not) is well known. The periodic scavenger task cleans these up.